roles/rancher/tasks/main.yml

```diff
@@ -705,14 +709,237 @@
     svc_access_line: "Login to any of these services as user 'admin', using the password you supplied."
   when: using_random_pwd is not defined
 
+- name: Login to KeyCloak
+  uri:
+    url: "https://{{ rancher_server }}/auth/realms/master/protocol/openid-connect/token"
+    method: POST
+    body_format: form-urlencoded
+    body:
+      client_id: admin-cli
+      grant_type: password
+      username: admin
+      password: "{{ rancher_password }}"
+    validate_certs: no
+    return_content: yes
+  register: kc_login_response
+  until: kc_login_response['status']
+  retries: 5
+  delay: 10
+
+- name: Set KeyCloak Token
+  set_fact:
+    kc_token: "{{ kc_login_response.json.access_token }}"
+
+- name: Create KeyCloak SAML Client
+  uri:
+    url: "https://{{ rancher_server }}/auth/admin/realms/master/clients"
+    method: POST
+    body_format: json
+    body:
+      clientId: "https://{{ rancher_server }}:{{ rancher_port }}/v1-saml/keycloak/saml/metadata"
+      name: "rancher"
+      protocol: "saml"
```

Comment:
So rancher doesn't support open id connect? That would be preferable over saml I think.

Reply:
https://github.com/rancher/rancher/issues/10053 This suggests that it's not yet supported and probably won't be in the near future